ENH: Initial Package Structure

hhslepicka opened the pull request:
This PR adds the basic package structure following the PCDS standards.
- GitHub Templates for `Issues` and `Pull Requests`
- Travis-CI support
- Documentation with Sphinx and automated deploy to `gh-pages` using doctr.
- Conda Recipe

teddyrendahl commented on .travis.yml:
```diff
@@ -0,0 +1,84 @@
+       env:
+          - BUILD=1
+          - PYDM_CHANNEL=pcds-tag
+     - python: 3.6
```
Any reason we can't do `3.7`?

hhslepicka replied:
PyDM is not yet full deployed/tested/packaged for 3.7...
I envision to add it in here soon...
`pyepics` is not yet available as a conda package for 3.7...
I will work on get it going with conda-forge.

teddyrendahl commented on .travis.yml:
```diff
@@ -0,0 +1,84 @@
+
+after_failure:
+  - cat logs/run_tests_log.txt
+  - cat logs/run_tests_log.txt.1    
```
Looks like you only run the tests once (which is fine, we only do that to make that we ship any non Python files correctly). However, you'll only have one test log

teddyrendahl commented on conda-recipe/meta.yaml:
```diff
@@ -0,0 +1,29 @@
+
+requirements:
+    build:
+      - python
```
`setuptools`?

teddyrendahl replied:
I think it largely doesn't matter (CONDA depends on setuptools), but I saw an issue if the requirements are specified in both `setup.py` and here in `happi`.

https://github.com/pcdshub/happi/pull/81

teddyrendahl commented on docs/source/conf.py:
```diff
@@ -0,0 +1,187 @@
+    'sphinx.ext.napoleon',
+    'sphinx.ext.autodoc',
+    'sphinx.ext.autosummary',
+    'IPython.sphinxext.ipython_directive',
```
Watch out. Some of these have a `matplotlib` requirement. I might delete the IPython ones until we need them (if ever)

teddyrendahl commented on github_deploy_key.enc:
```diff
@@ -0,0 +1 @@
+gAAAAABcCVrL7E7orKNCoUA9GqL3c-brRRUk4weQRJEGa0aGiCMg8OvxK7IZd34aSRPscyENa--HDEUv7qp6g6fEhQP9vND_kpnXme-sZh9wmFsuc4i48mOeZKCct_X30akuAQf2g2EK0zyGkh9LOEI2GY5ZMPyySzat5_Fyv8bxfi9VZ5_2Bc9aUvA3jnmk53DrVVn5jcYdygLBetTKRvnpSFxHfm58IGdH3fKhMpjvbAcNyrhAgfgW4Ju1oMbeTJZ-c30ffWyP6FUHmZaXCjRfzE3mvFnks0dehHR5x0TmWrhFFsaO70jzJRrA1ZBmX3Um0iL4ToNT7COfVrqbUoP_dip1DctXmSqcq4PqdLn_VXND7oblrA1x736pGIkWh3PN-_cjXbjvK-W ...
```
Don't redo this but I think you might be using an old version of `doctr`. Newer versions will name this key `github_deploy_key_pcdshub_pcdswidgets.enc` or something

hhslepicka commented:
@teddyrendahl Thank you for the review! I addressed your comments with the latest commits...
I added Python 3.7 to the matrix with the `allow_failures` for now.
